notebooks/2.modeling/nbconverted/modeling.py

```diff
@@ -0,0 +1,436 @@
+    ----------
+    model : BaseEstimator
+        best model
+    X : np.ndarray
+        features
+    y : np.ndarray
+        labels
+    shuffled : bool
+        Flag indicating if the data has been shuffled
+    seed : Optional[int], optional
+        _description_, by default 0
+
+    Returns
+    -------
+    tuple[pd.DataFrame]
+        returns a tuple that contains the f1 scores and precision/recall scores
+        in a dataframe
+    """
+
+    # setting seed
+    np.random.seed(seed)
+
+    # number of classes
+    n_classes = len(np.unique(y, axis=0))
+
+    # making predictions
+    predictions = model.predict(X)
+    probability = model.predict_proba(X)
+
+    # computing and collecting  precision and recall curve
+    precision_recall_scores = []
+    for i in range(n_classes):
+        # precision_recall_curve calculation
+        precision, recall, _ = precision_recall_curve(y[:, i], probability[:, i])
+
+        # iterate all scores and save all data into a list
+        for i in range(len(precision)):
+            precision_recall_scores.append([dataset, shuffled, precision[i], recall[i]])
+
+    # creating scores df
+    precision_recall_scores = pd.DataFrame(
+        precision_recall_scores, columns=["dataset", "shuffled", "precision", "recall"]
+    )
+
+    # Compute F1 score
+    f1_scores = []
+    for i in range(n_classes):
+        y_true = y[:, i]
+        y_pred = predictions[:, i]
+        f1 = f1_score(y_true, y_pred)
+        f1_scores.append([dataset, shuffled, injury_codes["decoder"][str(i)], f1])
+
+    # convert to data frame and display
+    f1_scores = pd.DataFrame(
+        f1_scores, columns=["data_set", "shuffled", "class", "f1_score"]
+    )
+
+    return (precision_recall_scores, f1_scores)
+
+
+# ## Training and Evaluating Multi-class Logistic Model with original dataset split
+#
+
+# In[7]:
+
+
+# train and get the best_model
+best_model = train_multiclass(X_train, y_train, param_grid=param_grid, seed=0)
+
+
+# In[ ]:
+
+
+test_precision_recall_df, test_f1_score_df = evaluate(
+    model=best_model, X=X_test, y=y_test, dataset="test", shuffled=False, seed=0
+)
+train_precision_recall_df, train_f1_score_df = evaluate(
+    model=best_model, X=X_train, y=y_train, dataset="train", shuffled=False, seed=0
+)
+
+
+# ## Training and Evaluating Multi-class Logistic Model with shuffled dataset split
+#
+
+# In[ ]:
+
+
+# setting random seed
+seed = 0
+np.random.seed(seed)
+
+# shuffle feature space
+shuffled_X_train = shuffle_features(X_train, seed=seed)
+
+
+# In[ ]:
+
+
+shuffled_best_model = train_multiclass(
+    shuffled_X_train, y_train, param_grid=param_grid, seed=seed
+)
+
+
+# In[ ]:
+
+
+shuffle_test_precision_recall_df, shuffle_test_f1_score_df = evaluate(
+    model=best_model, X=X_test, y=y_test, dataset="test", shuffled=True, seed=0
+)
+shuffle_train_precision_recall_df, shuffle_train_f1_score_df = evaluate(
+    model=best_model, X=X_train, y=y_train, dataset="train", shuffled=True, seed=0
+)
```

Comment:
```suggestion
shuffle_train_precision_recall_df, shuffle_train_f1_score_df = evaluate(
    model=best_model, X=shuffled_X_train, y=y_train, dataset="train", shuffled=True, seed=0
)
```

You need to evaluate on the data you trained on - a well controlled model has high training set performance, but low (close to random) performance in the test sest.